.travis.yml

```diff
@@ -0,0 +1,38 @@
+language: python
+python:
+  - "2.7"
+before_install:
+  - sudo apt-get update -qq
+  - sudo apt-get install mpi-default-bin mpi-default-dev
+
+  # LAMMPS
+  #  - build lammps as shared library
+  #  - and install LAMMPS python wrapper
+  - git clone git://git.lammps.org/lammps-ro.git mylammps
+  # checkout a recent stable version (from 28 June 2014)
+  - cd mylammps/src && git checkout r12164
+  # make shared library for python
+  - make makeshlib && make -f Makefile.shlib ubuntu_simple
+  - cd ../python && sudo python install.py /usr/lib/ $VIRTUAL_ENV/lib/python2.7/site-packages/
+  # make lammps executable and add to path
+  - cd ../src
+  - make ubuntu_simple
+  - ln -s lmp_ubuntu_simple lammps
+  - export PATH=$TRAVIS_BUILD_DIR/mylammps/src/:$PATH
+  - cd $TRAVIS_BUILD_DIR
+
+  # use current master of simphony
+  - pip install git+git://github.com/simphony/simphony-common.git#egg=simphony
```

Comment:
> I am not sure why we have to loop over all the CUBA keys? would it be better to loop over the existing CUBA in an object and only treat those, rather than looping over the predefined keys? is there a performace reason for this?

hmm. now that I am looking at the DataContainer code again I do not see any place where we loop over CUBA. I think I miss-red my own code where we check that a value is in CUBA. However that operation does not happen through a loop but through a hash function which should in theory not be affected by the size of the CUBA enum.  Sorry about  this.